solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -533,6 +535,47 @@ public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
     }
   }
 
+  public final <T extends SolrResponse> T requestWithBaseUrl(String baseUrl, SolrRequest<T> req)
+      throws SolrServerException, IOException {
+    return requestWithBaseUrl(baseUrl, req::process);
+  }
+
+  /**
+   * Temporarily modifies the client to use a different base URL and runs the provided lambda
+   *
+   * @param baseUrl the base URL to use on any requests made within the 'clientFunction' lambda
+   * @param clientFunction a Function that consumes a Http2SolrClient and returns an arbitrary value
+   * @return the value returned after invoking 'clientFunction'
+   * @param <R> the type returned by the provided function (and by this method)
+   */
+  public <R> R requestWithBaseUrl(String baseUrl, SolrClientFunction<SolrClient, R> clientFunction)
+      throws SolrServerException, IOException {
+    try (final var derivedClient = new NoCloseHttp2SolrClient(baseUrl, this)) {
```

Comment:
why try-finally here?     ;-)

Reply:
Two reasons, both dumb:

1. To prevent IDE's or static analysis from complaining that we're not closing a Closeable here.
2. Because `NoCloseHttp2SolrClient.close()` is mostly a no-op, but it does have one tiny bit of logic: it calls `ObjectReleaseTracker.release(this)` so that it doesn't get (falsely) picked up by the resource-leak tracking our tests do.

Reply:
fair enough; lets just add a one-liner excuse so this doesn't look so silly